scripts/checkpatch/typedefsfile

```diff
@@ -4,3 +4,5 @@ mbedtls_pk_context
 z_arch_esf_t
 pinctrl_soc_pin_t
 io_rw_32
+\b[A-Z]*_TypeDef
+\b[A-Z]*_[A-Za-z1-9_]*_TypeDef
```

Comment:
Noting that we can safely assume anything that is suffixed with `_TypeDef` will be typedefs, how about:
```suggestion
\b[a-zA-Z_][a-zA-Z0-9_]*_TypeDef
```
?